zwave_js_server/util/node.py

```diff
@@ -222,33 +222,55 @@ def _get_int_from_partials_dict(
     node: Node,
     partial_param_values: Dict[str, ConfigurationValue],
     property_: int,
-    new_value: Dict[int, Union[int, str]],
+    new_value: Dict[Union[int, str], Union[int, str]],
 ) -> int:
     """Take an input dict for a set of partial values and compute the raw int value."""
     int_value = 0
+    provided_partial_values = []
     # For each property key provided, we bit shift the partial value using the
     # property_key
-    for property_key, partial_value in new_value.items():
-        value_id = get_value_id(
-            node, CommandClass.CONFIGURATION, property_, property_key=property_key
-        )
-        if value_id not in partial_param_values:
-            raise NotFoundError(
-                f"Bitmask {property_key} ({hex(property_key)}) not found for "
-                f"parameter {property_}"
+    for property_key_or_name, partial_value in new_value.items():
+        # If the dict key is a property key, we can generate the value ID to find the
+        # partial value
+        if isinstance(property_key_or_name, int):
+            value_id = get_value_id(
+                node,
+                CommandClass.CONFIGURATION,
+                property_,
+                property_key=property_key_or_name,
             )
-        partial_value = _validate_and_transform_new_value(
-            partial_param_values[value_id], partial_value
-        )
-        int_value += partial_value << partial_param_bit_shift(property_key)
+            if value_id not in partial_param_values:
+                raise NotFoundError(
+                    f"Bitmask {property_key_or_name} ({hex(property_key_or_name)}) "
+                    f"not found for parameter {property_}"
+                )
+            zwave_value = partial_param_values[value_id]
+        # If the dict key is a property name, we have to find the value from the list
+        # of partial param values
+        else:
+            try:
+                zwave_value = next(
+                    value
+                    for value in partial_param_values.values()
+                    if value.property_name == property_key_or_name
+                )
+            except StopIteration:
+                raise NotFoundError(
+                    f"Partial parameter with label '{property_key_or_name}'"
+                    f"not found for parameter {property_}"
+                ) from None
+
+        provided_partial_values.append(zwave_value)
+        partial_value = _validate_and_transform_new_value(zwave_value, partial_value)
+        int_value += partial_value << partial_param_bit_shift(zwave_value.property_key)
```

Comment:
mypy: Argument 1 to "partial_param_bit_shift" has incompatible type "Union[str, int, None]"; expected "int"